### PR TITLE
Remove the intermediate truffle directory in lib

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java [hunk 1]
```diff
@@ -1729,7 +1729,7 @@ public RubyNode visitGlobalAsgnNode(GlobalAsgnParseNode node) {
             rhs = new CheckOutputSeparatorVariableTypeNode(rhs);
             rhs.unsafeSetSourceSection(sourceSection);
         } else if (name.equals("$_")) {
-            if (getSourcePath(sourceSection).endsWith(buildPartialPath("truffle", "rubysl", "rubysl-stringio", "lib", "rubysl", "stringio", "stringio.rb"))) {
+            if (getSourcePath(sourceSection).endsWith(buildPartialPath("rubysl", "rubysl-stringio", "lib", "rubysl", "stringio", "stringio.rb"))) {
```
Comment (eregon) [inline truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java]: (Unrelated) This seems a bit too much, let's hard-code the `/` as anyway that is what is visible at the Ruby level.

File: truffle/src/main/java/org/truffleruby/language/backtrace/BacktraceFormatter.java [hunk 2]
```diff
@@ -255,7 +255,7 @@ public static boolean isUserSourceSection(RubyContext context, SourceSection sou
             return false;
         }
 
-        return path.indexOf("/lib/ruby/stdlib/rubygems") == -1;
+        return path.indexOf("/lib/stdlib/rubygems") == -1;
```
Comment (eregon) [inline truffle/src/main/java/org/truffleruby/language/backtrace/BacktraceFormatter.java]: (Unrelated) Could be `!path.contains("...");`

File: truffle/src/main/ruby/core/rbconfig.rb [hunk 3]
```diff
@@ -79,21 +79,21 @@ module RbConfig
...
+    libdir = "#{ruby_home}/lib"
     bindir = "#{libdir}/bin"
 
     CONFIG.merge!({
       'bindir' => bindir,
       'libdir' => libdir,
-      "sitelibdir"=>"#{ruby_home}/lib/ruby/2.3/site_ruby", # TODO BJF Oct 21, 2016 Need to review these values
-      "sitearchdir"=>"#{ruby_home}/lib/ruby/2.3/site_ruby",
+      "sitelibdir"=>"#{ruby_home}/lib/2.3/site_ruby", # TODO BJF Oct 21, 2016 Need to review these values
+      "sitearchdir"=>"#{ruby_home}/lib/2.3/site_ruby",
```
Comment (eregon) [inline truffle/src/main/ruby/core/rbconfig.rb]: I would drop the `2.3` since it's not present anywhere else.

**Comment by chrisseaton** [comment]
@pitr-ch do you know what might be wrong with the tool since I removed the `ruby/truffle` part of the lib path?

**Comment by bjfish** [comment]
@chrisseaton This line might be causing the issue?
https://github.com/graalvm/truffleruby/blob/truffle-head/lib/ruby/truffle/jruby-truffle-tool/lib/truffle_tool.rb#L228

**Comment by pitr-ch** [comment]
The paths seem already fixed.

File: truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java (hunk 1 above)
Comment (nirvdrum) [inline truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java]: Also unrelated, but these sorts of string comparisons for path operations is the biggest reason we don't run on Windows. I realize it's not a going concern, but using the `java.nio.file.Path` API neatly avoids that.

Comment (nirvdrum) [inline truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java]: I do see that this code is just replacing the path being used. I'm just raising as FYI and partially in response to Benoit's suggestion.

**Pull request merged**